portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -14,34 +14,44 @@
 
 package com.google.sps.servlets;
 
-import java.io.IOException;
-import javax.servlet.annotation.WebServlet;
-import javax.servlet.http.HttpServlet;
-import javax.servlet.http.HttpServletRequest;
-import javax.servlet.http.HttpServletResponse;
-import java.util.ArrayList;
 import com.google.appengine.api.datastore.DatastoreService;
 import com.google.appengine.api.datastore.DatastoreServiceFactory;
 import com.google.appengine.api.datastore.Entity;
 import com.google.appengine.api.datastore.PreparedQuery;
 import com.google.appengine.api.datastore.Query;
 import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns that stores and retrieves comments from the Datastore. */
 
-/** Servlet that returns some example content. */
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+  // Creates a query that retrieves all the comment entities in the Datastore
+  Query QUERY = new Query("Comment").addSort("message", SortDirection.DESCENDING);
+
+  // Creates an instance of the Datastore so that comments can be retrieved, updated, and deleted
+  DatastoreService DATASTORE = DatastoreServiceFactory.getDatastoreService();
 
-  private ArrayList<String> comments = new ArrayList<>();
+  // Stores all the comment entites using the query defined
```

Comment:
nit: I was reading and found out that PreparedQuery doesn't actually store entities, but rather actually executes fresh every time you pull results out of it, e.g. `asIterable()` would cause it to fetch results. So I think something like "Gives access to all the comment entities" would be more accurate.